src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -393,6 +393,66 @@ class D2
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void VerifyExecutionOrder_TupleLiteralAndDeconstruction()
```

Comment:
>VerifyExecutionOrder_TupleLiteralAndDeconstruction [](start = 20, length = 50)

Were any behavior changes made for this scenario?

Reply:
No, this is to illustrate the evaluation order, as I was pushing some conversions down into tuple elements for tuple-equality like we did for deconstruction. Per our last conversation, tuple-equality should not push such conversions down.

---
In reply to: [171063660](https://github.com/dotnet/roslyn/pull/23356#discussion_r171063660) [](ancestors = 171063660)